### ISSUE TITLE
`soljson._compile is not a function` when using legacy solidity version

### ISSUE THREAD

**Issue by weiyuan95** [ethereum/solc-js#619]
Hi there,

I'm getting an error `soljson._compile is not a function` when trying to follow the steps here https://github.com/ethereum/solc-js#using-the-latest-development-snapshot . Here's a sample of the code i'm using. It's wrapped in a Promise since I want the callback to resolve before proceeding on to other parts of the code. The `solidityVersion` i'm calling this function with is `v0.4.18+commit.9cf6e910`, if it matters.

```
    const getCompiledContract = (solidityVersion: string): Promise<any> => {
      return new Promise((resolve, reject) => {
        solc.loadRemoteVersion(solidityVersion, (err: any, compiler: any) => {
          if (err) {
            return reject(err)
          } else {
            return resolve(compiler.compile(JSON.stringify(input)))
          }
        })
      })
    }
```
The error comes from when `compiler.compile(JSON.stringify(input))` is being run, and is triggered from `wrapper.ts`, line 341 in the source code, but I'm not sure what's the issue here.

Do let me know if there's any more information I can provide. Thanks!

**Comment by weiyuan95** [comment]
Running the code snipper below after manually downloading the binary works 
```
    let legacySolc = solc.setupMethods(require('./soljson-v0.4.18+commit.9cf6e910'))
    const contract = legacySolc.compile(JSON.stringify(input))
```
 So it seems like there's any issue with `.loadRemoteVersion`

**Comment by r0qs** [comment]
Hi @weiyuan95, what version of solc are you using? I tested your code here, and it compiles with solc version `0.8.17`, soljson `v0.4.18+commit.9cf6e910` and nodejs`18.9.0`. However, I get the following warning enabling `--trace-warnings` in nodejs:

```
(node:258003) V8: soljson-v0.4.18+commit.9cf6e910.js:3 Invalid asm.js: Invalid member of stdlib
    at Object.compileFunction (node:vm:360:18)
    at wrapSafe (node:internal/modules/cjs/loader:1048:15)
    at Module._compile (node:internal/modules/cjs/loader:1083:27)
    at IncomingMessage.<anonymous> (/home/roqs/ef/solcjs/bug619/node_modules/solc/wrapper.js:344:33)
    at IncomingMessage.emit (node:events:525:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

The warning does not happen when using the wasm build: https://binaries.soliditylang.org/wasm/soljson-v0.4.18+commit.9cf6e910.js
This warning is related to this issue: https://github.com/ethereum/solc-js/issues/537#issuecomment-881063273, and a possible explanation is here: https://github.com/trufflesuite/truffle/pull/4222#discussion_r689001206

In summary, in version 0.6.2 forward, the team switched to distribute wasm builds in https://binaries.soliditylang.org/bin instead of asm.js builds.
The `/bin` is currently the default location from where `loadRemoteVersion` retrieves the binaries. However, older binaries were not replaced, and you can still find the older versions in https://binaries.soliditylang.org/wasm (for newer versions, they are the same as in `/bin`).

**Comment by weiyuan95** [comment]
Thanks @r0qs ! I will close this, since this is quite outdated. Thanks for taking the time to take a look at this!

(If more people seem to experience this problem, i'l be more than happy to give more information)
